### ISSUE TITLE
TableView accessibility review issues

### ISSUE THREAD

**Issue by majornista** [adobe/react-spectrum#2132]
# 🐛 Bug Report 

1. "Select" and "Select All"  strings for labelling checkboxes should be localized.
2. In single selection mode, we set `visibility: hidden` on the Select All Checkbox, however, a `columnheader` must contain content to provide an accessibility name. Possible solutions are to either add a `VisuallyHidden` with "Select" as the content to the `columnheader`, or set the `role` for the `columnheader` to `gridcell`, when the Select All Checkbox is hidden.
3. Documentation examples include unlabelled icon Button controls for illustrative purposes. ideally we should demonstrate best practices for labelling these buttons.
4. Column header cell for `selectionMode="single"` should not have `aria-disabled="true"`.
5. With VoiceOver, NVDA and JAWS, nested column examples fail to announce nested columns correctly. For example, in [`/story/tableview--static-with-nested-columns`](https://reactspectrum.blob.core.windows.net/reactspectrum/924bb2f2b251bfe171cfd75440fbf9f0638b8836/storybook-17/iframe.html?id=tableview--static-with-nested-columns&providerSwitcher-locale=&providerSwitcher-theme=&providerSwitcher-scale=&providerSwitcher-toastPosition=bottom&viewMode=story), the first column after Select All, announces as "GROUP 1 TEST" and the next column announces as "GROUP 2 FOO" despite `aria-colspan` attributes indicating that the "TEST" column does not fall under a nested column heading, and that "FOO" and "BAR" should fall under the "GROUP 1" heading. This may be a screen reader/browser bug, reflecting a lack of support for `aria-colspan`:
   - https://bugs.webkit.org/show_bug.cgi?id=171165
   - https://bugs.chromium.org/p/chromium/issues/detail?id=543050
   - https://github.com/FreedomScientific/VFO-standards-support/issues?q=is%3Aissue+is%3Aopen+colspan
   - https://github.com/nvaccess/nvda/issues?q=is%3Aissue+is%3Aopen+colspan
6. Sort order change is not announced by screen readers when a sortable column header is clicked, and the current sort order state for the table cannot be determined when entering the table, without navigating through each column header. A suggested fix would be to use live region announcement to announce sort changes, and to use a `hidden` element referenced with `aria-describedby` to communicate the current sort order.

## 🔦 Context

These issues have been raised as part of an Accessibility review as we approach release for the TableView component.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | @react-spectrum/table@3.0.0-beta.1 
| Browser          | Chrome Version 91.0.4472.114 (Official Build) (x86_64)
| Operating System | MacOS 11.4 (20F71)

## 🧢 Your Company/Team

Adobe/Accessibility
